### ISSUE TITLE
Change single-line "rule" rules to be about single-line "declaration-block" instead

### ISSUE THREAD

**Issue by davidtheclark** [stylelint/stylelint#436]
_Breaking change_.

I think it's more more likely that a user wants, for example, `rule-no-single-line` to warn about this:

```css
.foo,
.bar,
.baz { color: pink; top: 0; bottom: 2px; }
```

... rather than letting it slide because there are line breaks in the selector.

Technically these rules are right the way they are because the "rule" includes the selector. But to make them more useful, I'd propose we change them to address the "declaration-block" instead of the complete rule.

This change would be breaking, and would include the following:

- [x] Replace rule-no-single-line with declaration-block-no-single-line
- [x] Replace rule-single-line-max-declarations with declaration-block-single-line-max-declarations

Because the changes are breaking let's reserve them for v3, grouping with other breaking changes.

@jeddy3 : Anymore you can think of?

**Comment by jeddy3** [comment]
> But to make them more useful, I'd propose we change them to address the "declaration-block" instead of the complete rule.

Sgtm.

> @jeddy3 : Anymore you can think of?

#437 makes me wondering whether we shouldn't use `3.0.0` as an opportunity to explicitly specify when things are attached to the `sheet`. 

* `no-duplicate-selectors` becomes `sheet-no-duplicate-selectors`
* `no-eol-whitespace` becomes `sheet-no-eol-whitespace`
* `no-missing-eof-newline` become `sheet-no-missing-eof-newline`
* `no-multiple-empty-lines` becomes `sheet-no-multiple-empty-lines`
* `indentation` becomes  `sheet-indentation`

At the moment those rules are kind of just floating around. Having them attached to the `sheet` will give us one convention across all the rule names, rather than the exception we've [had to document](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/rules.md#about-rule-names), and which might make it easier for new users to grasp the convention. I suspect they'll be more rules in the future that, like `sheet-no-duplicate-selectors`, apply to the whole sheet. What do you think?

I just looked through the other rules, and everything else looks good to me. I find it quite satisfying that we've successfully stuck to the convention across so many rules :)

**Comment by davidtheclark** [comment]
I might prefer to leave them floating (?) because they will not apply to specific stylesheets, but just to "all the CSS you fed me". If user has concatenated stylesheets or used postcss-import, the CSS they fed us is not a "stylesheet" in their mind. Does that make sense?

In #437 we might try to make an exception and throw in extra logic to make sure we're actually comparing selectors based on the source stylesheet; but that's not really relevant to these other rules.

Those are just some quick thoughts.

**Comment by jeddy3** [comment]
> Does that make sense?

Yep. Scratch the `sheet-*` suggestion :)

**Comment by jeddy3** [comment]
Done in #511
